File: app/src/main/java/com/neome/api/meta/base/dto/DefnFieldSetOfRole.java
// neome.ai API - do not change
//

package com.neome.api.meta.base.dto;

import com.neome.api.meta.base.Types.EnumDefnThemePickMultiVariant;
import com.neome.api.meta.base.Types.MetaIdField;
import com.neome.api.meta.base.Types.MetaIdRole;

import org.jetbrains.annotations.Nullable;

@SuppressWarnings("unused")
public class DefnFieldSetOfRole extends DefnFieldEditable
{
  @Nullable
  public Boolean allowSystemRoles;

  @Nullable
  public DefnStudioMapOfDtoOption callerRoleMap;

  @Nullable
  public MetaIdField defaultRoleFieldId;

  @Nullable
  public MetaIdRole[] defaultRoleIdSet;

  @Nullable
  public MetaIdRole[] excludeRoleIdSet;

  @Nullable
  public MetaIdRole[] filterRoleIdSet;

  @Nullable
  public DefnStudioMapOfDtoOption includeOptionMap;

  @Nullable
  public Long pageSize;

  @Nullable
  public EnumDefnThemePickMultiVariant showAs;
}

File: app/src/main/java/com/neome/api/meta/base/dto/DefnFieldSetOfRole.kt
// neome.ai API - do not change
//
@file:Suppress("unused", "MemberVisibilityCanBePrivate", "CanBeParameter", "PropertyName")


package com.neome.api.meta.base.dto

import com.neome.api.meta.base.dto.DefnFieldEditable
import com.neome.api.meta.base.dto.DefnStudioMapOfDtoOption
import com.neome.api.meta.base.Types.EnumDefnThemePickMultiVariant
import com.neome.api.meta.base.Types.MetaIdField
import com.neome.api.meta.base.Types.MetaIdRole

interface DefnFieldSetOfRole : DefnFieldEditable
{
  val allowSystemRoles: Boolean?
  val callerRoleMap: DefnStudioMapOfDtoOption?
  val defaultRoleFieldId: MetaIdField?
  val defaultRoleIdSet: List<MetaIdRole>?
  val excludeRoleIdSet: List<MetaIdRole>?
  val filterRoleIdSet: List<MetaIdRole>?
  val includeOptionMap: DefnStudioMapOfDtoOption?
  val pageSize: Long?
  val showAs: EnumDefnThemePickMultiVariant?
}